src/memory/memory.ts

```diff
@@ -101,12 +101,12 @@ export class Memory {
   private async summarize(events: Event[]): Promise<Event[]> {
     if (!events.length) return [];
 
-    const cumulativeTokenCounts = this.getCumulativeTokenCounts(events);
+    const cumulativeTokenCounts: number[] = this.getCumulativeTokenCounts(events);
```

Comment:
Bystander here. Generally it's considered preferable to use type inference based on the calls/code path, etc. E.g. here the method `getCumulativeTokenCounts` presumably always returns an array of numbers. For such situations one wouldn't generally add a type annotation to the receiver (const in this case) because it's redundant and it could make refactoring more cumbersome later.

Reply:
Noted, will take that one out then.

Reply:
Yeah, I think this comment basically applies to the whole PR. If TypeScript isn't able to infer a type it will create an error, so all of these are unnecessary. If you use a good code editor like VS Code you can easily inspect the types of any identifier by just hovering over it. I find type annotations everywhere to just visually clutter things without making anything safer.

Reply:
Totally fair, will take this one down and keep that in mind for future PRs.